lesson15/bilge/script.js

```diff
@@ -0,0 +1,182 @@
+
+
+console.log('Ex. 9:');
+console.log(domain);
+
+// 10. Replace Substring
+// Censor (replace 'bad' with 'good') a word in a sentence.
+const sentence = 'This is a bad example bad.';
+
+console.log('Ex. 10:');
+console.log(sentence.replace('bad', 'good'));
+
+// 11. Repeat a String (H)
+// Create a visual separator for console output. Expected result: "--------------------"
+const expression = 'hey';
+
+console.log('Ex. 11:');
+console.log(expression.repeat(3));
+
+// 12. Check if String Starts With (H)
+// Check if a file name starts with a specific prefix. Return true or false.
+const fileName = 'report_2024.pdf';
+
+console.log('Ex. 12:');
+console.log(fileName.startsWith('report'));
+
+// 13. Check if String Ends With (H)
+// Check if a file name ends with a specific extension. Return true or false.
+const extension = 'abc.pdf';
+
+console.log('Ex. 13:');
+console.log(extension.endsWith('.pdf'));
+
+// 14. Pad Start of String (H)
+// Pad a number with zeros using padStart() function to ensure it has at least 5 digits.
+const number = '42';
+
+console.log('Ex. 14:');
+console.log(number.padStart(5, '0'));
+
+// 15. Pad End of String (H)
+// Pad a text with spaces to align it to the right. Expected result: "Align     "
+const text = 'Align';
+
+console.log('Ex. 15:');
+console.log(text.padEnd(20), text); // I added the text variable next to the padded string to show it's aligned to the side.
+
+// 16. Split a String (H)
+// Split a sentence into words.
+const sentence2 = 'Learning JavaScript is fun';
+const splittedVersion = sentence2.split(' ');
+
+console.log('Ex. 16:');
+console.log(splittedVersion);
+
+// 17. Convert String to Array of Characters (H)
+// Convert a string into an array of its characters. Use sentence2 variable from exercise above.
+const convertToArray =  Array.from(sentence2);
+
+console.log('Ex. 17:');
+console.log(convertToArray);
+
+// 18. Reverse a String (H)
+// Reverse the characters in a string. Use sentence2 variable from exercise above.
+const splitSentence2 = sentence2.split('');
+const reversedSentence2 = splitSentence2.reverse();
+const joinedNewSentence2 = reversedSentence2.join('');
+
+console.log('Ex. 18:');
+console.log(joinedNewSentence2);
+
+// 19. Replace All Occurrences of a Substring (H)
+// Replace all instances of a word 'dog' in a text.
+const paragraph = "I think Ruth's dog is cuter than your dog!";
+
+console.log('Ex. 19:');
+console.log(paragraph.replaceAll('dog',  'cat'));
+
+// 20. Extract a Substring (H)
+// Extract the price and currency from a string below using "slice" method and print it to console. Should return "9.99 USD"
+const planPriceString = 'Premium plan - 9.99 USD/month.';
+const indexPriceString = planPriceString.indexOf('9.99 USD');
+
+console.log('Ex. 20:');
+console.log(indexPriceString); /* I used this variable which calculates the starting position of '9.99 USD' expression automatically without having the need to counting each letter manually */
```

Comment:
Good that you calculated the start index! You need to use it directly in the slice function. For the end index parameter, you can find it using start index and the length of the expression that you want to extract.

Reply:
Thanks Anna! I started with the same logic, but along the way I somehow got stuck in the end index; so I just used the start index.  Now I know how to do it. Shoul I go and change it and make a pull request again, or is it okay this way?

Reply:
You need to change in the current branch, commit and push. DO NOT create a new pull request or branch. 🙂

Reply:
I changed the solution with the one you suggested!